### PR TITLE
Document the build endpoint

### PR DESCRIPTION
**Pull request by danidoni**
This endpoint returns:

- the last build log 
- the build information
- the build status
- and the build history

for a specific artifact.

### CODE REVIEW

**Comment by danidoni** [comment]
[Build](https://trello.com/c/gsG0OEvj/1635-build)

File: src/api/public/apidocs-new/paths/build_project_repository_arch_package_log.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      example: "i586"
+      required: true
+      description: "The architecture short name"
+    - in: path
+      name: package
+      schema:
+        type: string
+      example: "ctris"
+      required: true
+      description: "The package name"
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/build_project_repository_arch_package_log.yaml:34]: There are 3 open PRs that define some combination of these parameters. Does it make sense to move them to their own files and add references to them whenever we need them?

I commented the same in PR #10976 and tried to do what I'm proposing in PR  #10975.
We need to agree regarding the names of the parameters: architecture VS architecture_name, for example.

**Review by saraycp — COMMENTED** [review]
LGTM. I just left a comment proposing to unify names and avoid duplicated common parameters definitions.

File: src/api/public/apidocs-new/components/parameters/repository_name.yaml [hunk 2]
```diff
@@ -3,5 +3,5 @@ name: repository_name
 schema:
   type: string
 required: true
-description: Repository name
+description: "Repository name"
```
Comment (saraycp) [inline src/api/public/apidocs-new/components/parameters/repository_name.yaml:6]: Is this change really needed? Should we add `""` to all the `*_name* parameters?

File: src/api/public/apidocs-new/paths/build_project_repository_arch_package_log.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - basic_authentication: []
+  parameters:
+    - $ref: '../components/parameters/project_name.yaml'
+    - $ref: '../components/parameters/repository_name.yaml'
+    - $ref: '../components/parameters/architecture_name.yaml'
+    - $ref: '../components/parameters/package_name.yaml'
+  responses:
+    '200':
+      description: |
+        This endpoint returns the log file contents
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/build_project_repository_arch_package_log.yaml:14]:
```suggestion
        This endpoint returns the log file content
```

File: src/api/public/apidocs-new/paths/build_project_repository_arch_package_history.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    - $ref: '../components/parameters/architecture_name.yaml'
+    - $ref: '../components/parameters/package_name.yaml'
+  responses:
+    '200':
+      description:
+        Returns ok when the build status is ok
+      content:
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/buildstatus.yaml'
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/build_project_repository_arch_package_history.yaml:18]: In this case, I think the schema is not `buildstatus` but `buildhistory`.

**Review by saraycp — CHANGES_REQUESTED** [review]
I have some little suggestions.

File: src/api/public/apidocs-new/components/parameters/repository_name.yaml (hunk 2 above)
Comment (danidoni) [reply src/api/public/apidocs-new/components/parameters/repository_name.yaml:6]: I'm affraid so. I'm preparing another PR that fixes several of those issues with our documented endpoints after passing some linter against them. If you want to check it by yourself: https://github.com/stoplightio/spectral

File: src/api/public/apidocs-new/paths/build_project_repository_arch_package_log.yaml (hunk 3 above)
Comment (danidoni) [reply src/api/public/apidocs-new/paths/build_project_repository_arch_package_log.yaml:14]: I think its `file contents` ([like here](https://stackoverflow.com/q/29881063))

Comment (danidoni) [reply src/api/public/apidocs-new/paths/build_project_repository_arch_package_log.yaml:14]: Well, I'm not sure. I'll accept your suggestion

File: src/api/public/apidocs-new/paths/build_project_repository_arch_package_history.yaml (hunk 4 above)
Comment (danidoni) [reply src/api/public/apidocs-new/paths/build_project_repository_arch_package_history.yaml:18]: So true

**Review by saraycp — APPROVED** [review]
